mysql-test/t/net_buffer_shrink_interval.test

```diff
@@ -0,0 +1,165 @@
+--echo Backup values
+set @old_net_buffer_length = @@global.net_buffer_length;
+set @old_net_buffer_shrink_interval = @@global.net_buffer_shrink_interval;
+
+--echo See the buffer size of the default connection
+--let $net_buffer_length= query_get_value(show status like 'net_buffer_length', Value, 1)
+--let $assert_text= net_buffer_length shall be 16384
+--let $assert_cond= $net_buffer_length = 16384
+--source include/assert.inc
+
+--echo Reduce initial buffer size for new connections
+set global net_buffer_length=4096;
+--let $net_buffer_length= query_get_value(show global variables like 'net_buffer_length', Value, 1)
+--let $assert_text= net_buffer_length shall be 4096
+--let $assert_cond= $net_buffer_length = 4096
+--source include/assert.inc
+
+--echo See the total usage before connection has been created
+--let $net_buffer_length= query_get_value(show global status like 'net_buffer_length', Value, 1)
+--let $assert_text= net_buffer_length shall be 16384
+--let $assert_cond= $net_buffer_length = 16384
+--source include/assert.inc
+
+set global net_buffer_shrink_interval=2;
+
+--source include/count_sessions.inc
+
+CREATE TABLE t1(data LONGBLOB) ENGINE=INNODB;
+
+#Create con_int1
+connect (con_int1,localhost,root,,);
+connection con_int1;
+
+--echo Check con_int1 buffer size, should be 4096
+--let $net_buffer_length= query_get_value(show status like 'net_buffer_length', Value, 1)
+--let $assert_text= net_buffer_length shall be 4096
+--let $assert_cond= $net_buffer_length = 4096
+--source include/assert.inc
+
+#Create con_int2
+connect (con_int2,localhost,root,,);
+connection con_int2;
+
+--echo Check con_int2 buffer size, should be 4096
+--let $net_buffer_length= query_get_value(show status like 'net_buffer_length', Value, 1)
+--let $assert_text= net_buffer_length shall be 4096
+--let $assert_cond= $net_buffer_length = 4096
+--source include/assert.inc
+
+--echo See the total usage after connections have been created
+--let $net_buffer_length= query_get_value(show global status like 'net_buffer_length', Value, 1)
+--let $assert_text= net_buffer_length shall be 24576
+--let $assert_cond= $net_buffer_length = 24576
+--source include/assert.inc
+
+#Insert big value(8096 bytes) to increase buffers in both connections
+connection con_int1;
+
+INSERT INTO t1 (data) VALUES ("
```

Comment:
To avoid ugly-looking long strings you can create an MTR variable
```
--let $long_string = `SELECT REPEAT('EF', <big_number>)`
```
and then use it in the `INSERT` statement
```
eval INSERT INTO t1 (data) VALUES ("$long_string");
```

Reply:
good idea @percona-ysorokin , I will do